### ISSUE TITLE
"HTTP 414 Request-URI Too Long" when posting a large note

### ISSUE THREAD

**Issue by jfromaniello** [jenkinsci/gitlab-plugin#455]
## Issue

It fails to post a customized note when it is too large.

### Context
- **Gitlab plugin version**: 1.4.1.1
- **Gitlab version**: 8.14
- **Jenkins version**: 2.34

### Logs & Traces

The error shown in jenkins is:

```
Failed to add comment on Merge Request for project '119': HTTP 414 Request-URI Too Long
Failed to add vote on Merge Request for project '119': HTTP 404 Not Found
```

I am not familiar with Java nor the framework used for sending requests but it seems that the plugin is sending the "body" of the NOTE on [createMergeRequestNote](https://github.com/jenkinsci/gitlab-plugin/blob/bc73655530e0aae71f4b0747e099e0d3c10ca8e0/src/main/java/com/dabsquared/gitlabjenkins/gitlab/api/GitLabApi.java#L107-L109) as a "query" string parameter?

If that's the case my suggestion send body as the actual body.

**Comment by jfromaniello** [comment]
@coder-hugo thanks a lot!

**Comment by omehegan** [comment]
This was released in version 1.4.4 of the plugin.
